### ISSUE TITLE
Mouse events getting lost from dynamically bound list in Chrome  in some cases

### ISSUE THREAD

**Issue by willseitz** [angular/angular#21779]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

When I have the following conditions I am unable get get events to fire:
1. I am binding an array to a component with a ul and an *ngFor on the lis inside.
2. I have both (mouseover) and (mousedown) events on the li
3. The elements of the array are objects (works for strings)
4. I am binding to a function that returns a new array instead of the same array every time
5. I am using Chrome (this works on Firefox, Edge, and even IE11!?!?)

My Plunkr is at https://plnkr.co/edit/4ywHbvutSVQy8rQu1B4h?p=preview. Clicking on elements in the top list work while clicking on those in the bottom fail. Removing the mouseover event will make it work. Using another browser makes this work (maybe this is a Chrome issue). The essential problem seems to be the combination of the dynamic list and mouseover events causes the ul to be recreated in the DOM although it does not need to be. This seems to interfere with the existing mouse events.

## Expected behavior

Both lists should be able to fire events.

## Minimal reproduction of the problem with instructions

https://plnkr.co/edit/4ywHbvutSVQy8rQu1B4h?p=preview

## What is the motivation / use case for changing the behavior?

I am trying to build a dropdown list that binds to a list of options that may change. Hovering over and clicking the events is required.

## Environment

<pre><code>
Angular version: 5.2.2

Browser:
- [X] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by robwormald** [comment]
Your demo is basically written to cause an infinite loop, as your `dynamicList()` function returns a *new instance* on every change detection run, which will trigger the entire list to re-render, which will construct a new element, then trigger a mouseover event, which will trigger a new CD run, which will trigger the new list...

You should basically avoid binding to methods - angular *must* re-execute that method to detect if changes have been made. Just don't do it.

Additionally, adding the `trackBy` parameter to the `ngFor` prevents the list from being entirely re-rendered (normally, it tracks by identity, but your `dynamicList` method provides a whole new set of identities)

I suspect this is specific to Chrome due to timing more than anything else...

**Comment by willseitz** [comment]
This demo does not seem to cause an infinite loop. It works in IE11, Firefox, and Edge and does not appear to loop infinitely. If Angular is catching an infinite loop condition behind the scenes it should probably throw an exception like the one it throws it dev mode where the last change detection cycle detects inconsistent data.

I thought binding to code was a common practice like `[class.selected]="hero === selectedHero"` from the tutorial. 

I found that the behavior was very consistent, so it didn't seem like a timing issue or at least was a very deterministic one. One event handler would be attached and work while the other didn't in a specific browser.  Perhaps when the event was fired after being attached to the new element is different between browsers causing one browser to attach the mousedown event successfully while other didn't. This is not a blocking issue, but rather something I encountered during progressive refactoring that confused me. The condition to create this bug seemed very precise so it felt like an edge condition failure.
